### PR TITLE
Update dependencies

### LINKED ISSUE #65: Fix the SSL certificates in the tests
Currently the tests need the setting `GODEBUG=x509ignoreCN=0`.  That's because without that, on newer versions of go, the tests give the following error:

```
Post \"https://localhost:12984/access-token\": x509: certificate relies on legacy Common Name field, use SANs or temporarily enable Common Name matching with GODEBUG=x509ignoreCN=0
```

So ideally, the test suite needs to be updated so that it provisions certs that work with the expectations of the new versions of go (using SAN instead of CN, presumably).  Then we can remove the GODEBUG workaround from the makefile.

### PR DESCRIPTION
**Pull request by zemanlx**
Fixes vulnerable dependencies and as a bonus, it fixes #65.

### CODE REVIEW

File: e2e/ssltest/certificates.go [hunk 1]
```diff
@@ -29,6 +29,7 @@ func CreateCertificates(cn, destDir string) (certFile, keyFile string) {
 			Subject: pkix.Name{
 				CommonName: cn,
 			},
+			DNSNames: []string{cn},
```
Comment (aecay) [inline e2e/ssltest/certificates.go:32]: Does this change solve https://github.com/sky-uk/osprey/issues/65?

Comment (zemanlx) [reply e2e/ssltest/certificates.go:32]: Yes

Comment (aecay) [reply e2e/ssltest/certificates.go:32]: Wonderful :tada: I think this change lgtm but we will probably want a second opinion from someone who is more experienced with the osprey.  @howardburgess has reviewed osprey changes for me before so maybe he can do the honors...

Comment (howardburgess) [reply e2e/ssltest/certificates.go:32]: LGTM! (Sorry for delay, was on hols)

**Review by howardburgess — APPROVED** [review]

**Pull request merged**